kornia/augmentation/functional.py

```diff
@@ -465,3 +482,42 @@ def apply_affine(input: torch.Tensor,
         return out_data.view_as(input), transform
 
     return out_data.view_as(input)
+
+
+def apply_rotation(input: torch.Tensor, params: Dict[str, torch.Tensor], return_transform: bool = False):
+    r"""Rotate a tensor image or a batch of tensor images a random amount of degrees.
+    Input should be a tensor of shape (C, H, W) or a batch of tensors :math:`(*, C, H, W)`.
+
+    Args:
+        params (dict): A dict that must have {'degrees': torch.Tensor}. Can be generated from
+                       kornia.augmentation.param_gen._random_rotation_gen
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+                                      input tensor. If ``False`` and the input is a tuple the applied transformation
+                                      wont be concatenated
+    """
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    device: torch.device = input.device
+    dtype: torch.dtype = input.dtype
+
+    input = input.unsqueeze(0)
+    input = input.view((-1, (*input.shape[-3:])))
+    angles: torch.Tensor = params["degrees"].to(device, dtype)
+
+    transformed: torch.Tensor = rotate(input, angles).squeeze(0)
+
+    if return_transform:
+
+        center: torch.Tensor = _compute_tensor_center(input)
+        rotation_mat: torch.Tensor = _compute_rotation_matrix(angles, center.expand(angles.shape[0], -1))
```

Comment:
another option could be calling get_rotation_matrix2d + warp_perspective. For now it's fine